src/main/scala/org/mdedetrich/apache/sonatype/ApacheSonatypeKeys.scala

```diff
@@ -30,5 +30,6 @@ trait ApacheSonatypeKeys {
     settingKey[File]("The NOTICE file which needs to be included in published artifact")
   lazy val apacheSonatypeDisclaimerFile: SettingKey[Option[File]] =
     settingKey[Option[File]]("The DISCLAIMER file which can optionally be included in published artifact")
-
+  lazy val apacheSonatypeArtifactNamePrefix: SettingKey[Option[String]] =
+    settingKey[Option[String]]("A prefix that will be added before the artifact name")
```

Comment:
Maybe we should just make this a function from the artifact id?

```scala
lazy val apacheSonatypeArtifactNameGenerator: SettingKey[String => String] =
 settingKey[String => String]("A function to generate a display name for the artifact given the artifact ID")
```

Then we could define it like:

```scala
apacheSonatypeArtifactNameGenerator :=
  artifactId => s"Apache ${artifactId.split('-').map(_.capitalize).mkString(" ")}"
```

Reply:
If we need to pretty print, I was thinking of just making a pretty print function (which using your example would just be `artifactId.split('-').map(_.capitalize).mkString(" ")`).

Reason being that if you were to change these settings, it would be one or the other (i.e. prefix or prettifier). Its unlikely that you would have different prefix's/prettifiers per project/sbt module and they are also different concetps.

I am not dead set against this though, lets see what the other feedback is.

Reply:
Incase my previous point isn't clear, since this is meant to be a general sbt plugin that isn't necessarily tied to Pekko, it could be that some other Apache project using this plugin has set their sbt `name := apache-myproject` in which case you would set `apacheSonatypeArtifactNamePrefix := None` and `apacheSonatypeArtifactNamePrettifier` would remain unchanged.

Reply:
Provides the generator with a default setting seems great.

Reply:
> Yeah, I was thinking about this. If this plugin is meant for Apache projects, and as a rule the name must always start with `Apache`, should we just check and add the prefix if it isn't already there?
> 
> ```scala
> apacheSonatypeArtifactNameGenerator := { artifactId =>
>   val prettified = artifactId.toLowerCase.split('-').map(_.capitalize).mkString(" ")
>   if (prettified.startsWith("Apache ")) prettified else s"Apache $prettified"
> }
> ```
> 
> That seems like it would cover 99% of the cases users would care about.
> 
> Anyway, I don't feel too strongly about this, I just feel like a single setting is simpler as long as we can make it do the right thing in most cases.

So I ended up going with this suggestion, just using `apacheSonatypeArtifactNameProcessor` as a more suitable name for the key